code/datums/loot_spawner.dm

```diff
@@ -142,79 +149,124 @@
 			if(initial(A.spawn_blacklisted))
 				blacklist_paths_data << "[path]"
 
+/*get_spawn_value()
+this proc calculates the spawn value of the objects based on factors such as
+their frequency, their rarity value, their price and
+the data returned by the get_special_rarity_value() proc
+*/
 /datum/controller/subsystem/spawn_data/proc/get_spawn_value(npath)
 	var/atom/movable/A = npath
-	var/spawn_value = 10 * initial(A.spawn_frequency)/(initial(A.rarity_value) + log(10,max(get_spawn_price(A),1)))
-	return spawn_value
+	if(ispath(npath, /obj/item/weapon/gun))
+		return 10 * initial(A.spawn_frequency)/(get_special_rarity_value(npath)+(get_spawn_price(A)/GUN_PRICE_DIVISOR))
+	else if(ispath(npath, /obj/item/clothing))
+		return 10 * initial(A.spawn_frequency)/(get_special_rarity_value(npath) + (get_spawn_price(A)/CLOTH_PRICE_DIVISOR))
+	return 10 * initial(A.spawn_frequency)/(get_special_rarity_value(npath) + log(10,max(get_spawn_price(A),1)))
+
+/*get_special_rarity_value()
+increases the rarity value of items
+depending on certain determining factors,
+for example, the rarity value of power cells increases with their max_charge,
+the value of stock parts increases with the rating.
+*/
+/datum/controller/subsystem/spawn_data/proc/get_special_rarity_value(npath)
+	var/atom/movable/A = npath
+	. = initial(A.rarity_value)
+	if(ispath(npath, /obj/item/weapon/cell))
+		var/obj/item/weapon/cell/C = npath
+		var/bonus = 0
+		var/autorecharging_factor = 3.7
+		if(ispath(npath, /obj/item/weapon/cell/large))
+			bonus += (initial(C.maxcharge)/CELL_LARGE_BASE_CHARGE)**1.2
```

Comment:
The one issue I can see with this, but not one that can be fixed without a major overhaul, is that aged cells have a reduced max charge, so would presumably be worth less.
I'm sure that's covered somewhere else in the code, though.

Reply:
the items are aged after creating the loot (after the object is instantiated) and their probability of appearance is determined elsewhere, this only takes into account the initial value of the paths.